### PR TITLE
XDR-1276: Add OOB Sentinel ARM data connectors

### PR DESCRIPTION
**Pull request by chrisjaimon2012**

### CODE REVIEW

**Review by arledesma — DISMISSED** [review]
Great first pass.

Let's make sure to provide useful descriptions and expose our variables in `snake_case`.

We will need to do our best to not output secrets.  If we do require that to exist in the output then lets talk about it -- we may need to address the upgrade to `1.1`, as we do not get access to sensitive until at least [`0.15`](https://www.terraform.io/language/functions/sensitive).

Thank you for using `terraform-docs` to generate the README for each of the modules.  We will be incorporating that into our pipeline shortly as these modules are open sourced.

File: modules/sentinel-data-connector-confluence-audit/vars.tf [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+variable "resource_group_name" {
+  description = "The name of the Resource Group where the Resource Group Template Deployment should exist."
+  type        = string
+}
+
+variable "arm_template" {
+  description = "The ARM template URL to download the template from."
+  type        = string
+  default     = "https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/DataConnectors/AtlassianConfluenceAudit/azuredeploy_Connector_ConfluenceAuditAPI_AzureFunction.json"
```
Comment (lukiffer) [inline modules/sentinel-data-connector-confluence-audit/vars.tf:18]: We want our infrastructure modules to be idempotent. Referencing the upstream `master` branch here introduces an element of variability that is undesirable. We should pin this to a specific commit hash.

**Review by lukiffer — CHANGES_REQUESTED** [review]
Found a few things to change that I missed during the initial glance. Apologies for the delay in review.

File: modules/sentinel-data-connector-confluence-audit/vars.tf (hunk 1 above)
Comment (chrisjaimon2012) [reply modules/sentinel-data-connector-confluence-audit/vars.tf:18]: Will do

File: modules/sentinel-data-connector-arm-generic/main.tf [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  required_providers {
+    azurerm = ">= 2.54.0"
+  }
+}
+
+data "http" "arm_template" {
+  url = var.arm_template
+}
+
+resource "azurerm_resource_group_template_deployment" "data_connector_arm" {
```
[the remark this answers is not in the document]
Comment (chrisjaimon2012) [reply modules/sentinel-data-connector-arm-generic/main.tf:16]: The module uses [azurerm_template_deployment](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/template_deployment) and I've used [azurerm_resource_group_template_deployment](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/resource_group_template_deployment)

`azurerm_template_deployment` has some limitations like
- doesn't clean up ARM resources during terraform destroy
- doesn't support parameters or outputs of non-string values
- more [here](https://github.com/hashicorp/terraform-provider-azurerm/pull/7506#issuecomment-700531519)

Should I leave it as a resource block for now? Or can I go ahead and update [our module](https://github.com/quantum-sec/package-azure/blob/master/modules/azure-resource-manager-template/main.tf#L9) from `azurerm_template_deployment` to `azurerm_resource_group_template_deployment`

[the remark this answers is not in the document]
Comment (lukiffer) [reply modules/sentinel-data-connector-arm-generic/main.tf:16]: Oh, that's interesting. I didn't catch that. Let's keep this as-is for now so as not to break backwards compatibility with the other module. And we can create a follow-up story to look into migrating that module.

File: modules/sentinel-data-connector-confluence-audit/vars.tf [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+variable "resource_group_name" {
+  description = "The name of the Resource Group where the Resource Group Template Deployment should exist."
+  type        = string
+}
+
+variable "arm_template" {
+  description = "The ARM template URL to download the template from."
+  type        = string
+  default     = "https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/DataConnectors/AtlassianConfluenceAudit/azuredeploy_Connector_ConfluenceAuditAPI_AzureFunction.json?ref=9218a88"
```
Comment (lukiffer) [inline modules/sentinel-data-connector-confluence-audit/vars.tf:18]: I don't think that the `?ref=XYZ` query string is doing anything.

I think the URL scheme is `https://raw.githubusercontent.com/Azure/Azure-Sentinel/<FULL_COMMIT_HASH>/DataConnectors/AtlassianConfluenceAudit/azuredeploy_Connector_ConfluenceAuditAPI_AzureFunction.json`

**Review by lukiffer — APPROVED** [review]
👍

**Pull request merged**